### PR TITLE
feat: improve folder structure, add hex and plugin folders

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Group functions in hex and plugin folders. Add tests for all functions
 
Task ID: [OS-939](https://aragonassociation.atlassian.net/browse/OS-939)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

[OS-1006]: https://aragonassociation.atlassian.net/browse/OS-1006?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OS-939]: https://aragonassociation.atlassian.net/browse/OS-939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sdk/src/hex/bytes.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Encodes a buffer into a hex string with the "0x" prefix
+ *
+ * @export
+ * @param {string} hexString
+ * @return {Uint8Array}
+ */
+export function hexToBytes(hexString: string): Uint8Array {
+  if (!hexString) return new Uint8Array();
```
Comment (jordaniza) [inline sdk/src/hex/bytes.ts:12]: If the possibility of hexString is to be a non-falsy string value then the param should reflect that. Else explicitly check for  `=== ''`

File: sdk/src/plugin/proposal.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  const matchedRegexResult =
+    proposalId.match(OSX_COMPACT_PROPOSAL_ID_REGEX) || [];
+  if (matchedRegexResult.length !== 3) {
+    throw new InvalidProposalIdError(proposalId);
+  }
+
+  return {
+    pluginAddress: matchedRegexResult[1],
+    id: parseInt(strip0x(matchedRegexResult[2]), 16),
+  };
```
Comment (jordaniza) [inline sdk/src/plugin/proposal.ts:52]: Might be useful to rename `3` and the indicies as const values to indicate what these are

**Review by jordaniza — COMMENTED** [review]
Good pr, makes sense. Can approve shortly once you've resolved comments

**Review by jordaniza — DISMISSED** [review]

File: sdk/src/plugin/proposal.ts (hunk 2 above)
Comment (Rekard0) [reply sdk/src/plugin/proposal.ts:55]: or at least add comment explaining the indices

File: sdk/test/unit/plugin/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  encodeRatio,
+} from '../../../src';
+
+describe('plugin/utils', () => {
+  describe('encodeRatio', () => {
+    it('Should encode a bigint from a float and the number of digits', () => {
+      const inputs = [
+        {float: 0.5, digits: 1, out: BigInt(5), error: null},
+        {float: 1, digits: 4, out: BigInt(10000), error: null},
+        {float: 0.25555, digits: 2, out: BigInt(26), error: null},
```
Comment (Rekard0) [inline sdk/test/unit/plugin/utils.test.ts:14]: the naming is a bit confusing in general in these tests
some are named `const inputs` & some are named `const tests`
some object has `output` and some has `out`

i suggest creating a type of needed or at least follow a consistent structure:

test = { input, output, error }

**Review by Rekard0 — COMMENTED** [review]
The tests are good, nice work,
but they need some consistency, i will re-review once they are ready

**Review by jordaniza — APPROVED** [review]

**Review by Rekard0 — DISMISSED** [review]

**Review by jordaniza — APPROVED** [review]

**Review by jordaniza — COMMENTED** [review]
Why the force push?

File: sdk/src/context-core.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import {GRAPHQL_NODES, IPFS_NODES, LIVE_CONTRACTS} from './constants';
```
Comment (Rekard0) [inline sdk/src/context-core.ts:1]: rethink file name

**Review by Rekard0 — APPROVED** [review]

File: sdk/src/context-core.ts (hunk 4 above)
Comment (josemarinas) [reply sdk/src/context-core.ts:1]: is going to get deleted, this will be legacy code with the new sdk

**Pull request merged**